controls/cf_serverd.cf

```diff
@@ -42,15 +42,31 @@ body server control
       allowusers            => { "root" };
 
     windows::
-      cfruncommand => "$(sys.cf_agent) -I -D cfruncommand -f \"$(sys.update_policy_path)\"  &
-                       $(sys.cf_agent) -I -D cfruncommand";
+      cfruncommand => "$(sys.cf_agent) -I -D cf_runagent_initiated -f \"$(sys.update_policy_path)\"  &
+                       $(sys.cf_agent) -I -D cf_runagent_initiated";
 
-    !windows::
-      # last single quote in cfruncommand is left open, so that
-      # arguments (like -K and --remote-bundles) are properly appended.
-      cfruncommand => "$(def.cf_runagent_shell) -c \'
-                           $(sys.cf_agent) -I -D cfruncommand -f $(sys.update_policy_path)  ;
-                           $(sys.cf_agent) -I -D cfruncommand";
+    !windows.!cfengine_3_7::
+
+      # In 3.10 the quotation is properly closed when EOF is reached. It is left
+      # open so that arguments (like -K and --remote-bundles) can be appended.
+      # 3.10.x does not automatically append -I -Dcfruncommand
+
+        cfruncommand => "$(def.cf_runagent_shell) -c \'
+                           $(sys.cf_agent) -I -D cf_runagent_initiated -f $(sys.update_policy_path)  ;
+                           $(sys.cf_agent) -I -D cf_runagent_initiated";
```

Comment:
Instead of introducing a new class name because of 3.7.x, I would rather keep the class named `cfruncommand` and honour the legacy. :-) Specifically for `update.cf` I would completely remove `-D` argument for 3.7.x.

Changing the class name also nullifies the 3.7.x "protection" of refusing to run other policy files, so anybody will be able to append arbitrary arguments (and potentially `-f` argument).

If you change the class name after all, ensure a loud changelog message and a section in the release notes.

Reply:
This still doesn't make sense to me, @jimis.  The options go from `cf-runagent` (on the hub) to `cf-serverd` (on the client) and from there to `cf-agent` on the client.  If `cf-serverd` is version 3.7, it will append `-I -D cfruncommand` to the arguments it passes to `cf-agent` regardless of what's in masterfiles, so it doesn't have to be in policy.  And if `cf-serverd` is version 3.10, then presumably `cf-agent` is also 3.10 (they're on the same host) and in that case `-D cfruncommand` doesn't cause any special behavior in the agent at all, right?

So in what circumstances will omitting `-D cfruncommand` from *policy* allow arbitrary arguments to be passed via `cf-runagent` through `cf-serverd` to `cf-agent` and be acted upon?

Reply:
You are right @mikeweilgart so the danger is not there as I initially thought. I withdraw my objection. :+1: